service/controller/rbac/selector.go

```diff
@@ -0,0 +1,20 @@
+package rbac
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type wrongSelector struct {
```

Comment:
Why is this called `wrongSelector`?

Reply:
I'm not happy with the implementation. I think it's wrong because it's based on the assumption that operatorkit only uses `Matches`. If the underlying implementation changes this may result in a  panic. In my opinion we should change the interface in operatorkit. I'll add a comment here.

Reply:
I agree. I'm the one that made this change in https://github.com/giantswarm/operatorkit/pull/512. We used to have an interface in `operatorkit`, but we needed to use server-side filtering to fix that bug and that only accepted a `labels.Selector`.

We seem to be moving away from `operatorkit` so maybe it doesn't have to be perfect.

Reply:
Not a fan of the name. If the code is used in production then it's not wrong. But it's fine if you want to keep it this way.